src/opts.rs

```diff
@@ -33,6 +33,29 @@ where
     Ok(())
 }
 
+/// Set the memory allocator to a different memory allocator.
+/// 
+/// This allocator will then be used to make all memory allocations for
+/// libgit2 operations. If the given `allocator` is None, then the
+/// system default will be restored.
+pub unsafe fn set_allocator(
```

Comment:
For an unsafe fn, we'll need to put a comment here for callers to know how to use it safely

Reply:
Alternatively, we don't provide this unsafe binding. People can use libgit2-sys directly

Reply:
Yeah, that's a fair point. Would it be better to take an `std::alloc::Allocator` as an input?